packages/flutter/lib/src/widgets/async.dart

```diff
@@ -737,6 +743,17 @@ class FutureBuilder<T> extends StatefulWidget {
   /// [AsyncSnapshot.hasError] will be true.)
   final T? initialData;
 
+  /// The builder used when the [AsyncSnapshot] is in the [ConnectionState.waiting] state.
+  final WidgetBuilder? onPending;
+
+  /// The builder used when the [AsyncSnapshot] has an error.
+  final WidgetBuilder? onError;
+
+  /// The listener that is called when the passed in [Future] state changes.
```

Comment:
Why does this have to be part of the widget? You can just listen to the future directly?